assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -0,0 +1,367 @@
+				sharedRoles: [ 'editor' ],
+				management: 'all_admins',
+			},
+		},
+	};
+
+	let registry;
+	let store;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		store = registry.stores[ CORE_MODULES ].store;
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setSharingManagement', () => {
+			const settingsWithoutManagement = {
+				'search-console': {
+					sharedRoles: [ 'editor', 'subscriber' ],
+				},
+				analytics: {
+					sharedRoles: [ 'editor' ],
+				},
+				'pagespeed-insights': {
+					sharedRoles: [ 'editor' ],
+				},
+			};
+
+			it( 'requires the moduleSlug param', () => {
+				expect( () => {
+					registry.dispatch( CORE_MODULES ).setSharingManagement();
+				} ).toThrow( 'moduleSlug is required' );
+			} );
+
+			it( 'requires the management param', () => {
+				expect( () => {
+					registry
+						.dispatch( CORE_MODULES )
+						.setSharingManagement( 'analytics' );
+				} ).toThrow( 'management must be one of: all_admins, owner.' );
+			} );
+
+			it( 'receives management and sets it to the sharing settings modules', () => {
+				registry
+					.dispatch( CORE_MODULES )
+					.receiveGetSharingSettings( settingsWithoutManagement );
+
+				const state = {
+					...store.getState().sharingSettings,
+					...store.getState().savedSharingSettings,
+				};
+
+				const moduleSlugs = [ 'analytics', 'search-console' ];
+				const managementRoles = [ 'all_admins', 'owner' ];
+
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement(
+						moduleSlugs[ 0 ],
+						managementRoles[ 0 ]
+					);
+				registry
+					.dispatch( CORE_MODULES )
+					.setSharingManagement(
+						moduleSlugs[ 1 ],
+						managementRoles[ 1 ]
+					);
+
+				const sharingSettingsWithManagement = moduleSlugs.reduce(
+					( sharingSettingsObj, moduleSlug, index ) => ( {
+						...sharingSettingsObj,
+						[ moduleSlug ]: {
+							...settingsWithoutManagement[ moduleSlug ],
+							management: managementRoles[ index ],
+						},
+					} ),
+					{}
+				);
+
+				expect( store.getState().sharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettingsWithManagement,
+				} );
+			} );
+		} );
+
+		describe( 'setSharedRoles', () => {
+			const settingsWithoutRoles = {
+				'search-console': {
+					management: 'all_admins',
+				},
+				analytics: {
+					management: 'owner',
+				},
+				'pagespeed-insights': {
+					management: 'all_admins',
+				},
+			};
+
+			it( 'requires the moduleSlug param', () => {
+				expect( () => {
+					registry.dispatch( CORE_MODULES ).setSharedRoles();
+				} ).toThrow( 'moduleSlug is required' );
+			} );
+
+			it( 'requires the roles param', () => {
+				expect( () => {
+					registry
+						.dispatch( CORE_MODULES )
+						.setSharedRoles( 'analytics' );
+				} ).toThrow( 'roles must be an array of strings.' );
+			} );
+
+			it( 'receives roles and sets it to the sharing settings modules', () => {
```

Comment:
Same here. No need to overcomplicate it. Just set the original settings, dispatch the `setSharedRoles` action for one module and check that the appropriate value is set in the state.